neuralop/layers/spectral_convolution.py

```diff
@@ -457,14 +461,26 @@ def forward(
         # otherwise drop first two dims (in_channels, out_channels)
         else:
             weight_start_idx = 2
-        starts = [(size - min(size, n_mode)) for (size, n_mode) in zip(list(x.shape[2:]), list(weight.shape[weight_start_idx:]))]
+        print(f"{weight.shape=}")
+        starts = [(size - min(size, n_mode)) for (size, n_mode) in zip(fft_size, list(weight.shape[weight_start_idx:]))]
         slices_x =  [slice(None), slice(None)] # Batch_size, channels
 
+        # when x is even length along any dimension and we select an odd number of modes,
+        # torch.fft.fftshift will place the modes off-center and we prioritize negative frequencies by default
+        # we add a simple check which shifts the indices forward by 1 along each dim where this is the case
+
+        start_shifts = [1 if size % 2 == 0 and modes % 2 == 1 else 0 for size, modes in zip(fft_size, list(weight.shape[weight_start_idx:]))]
+
         if self.complex_data:
-            slices_x += [slice(start//2, -start//2) if start else slice(start, None) for start in starts]
+            slices_x += [slice(start//2 + shift, -start//2  + shift) if start else slice(start, None) for start, shift in zip(starts, start_shifts)]
         else:
-            slices_x += [slice(start//2, -start//2) if start else slice(start, None) for start in starts[:-1]]
+            slices_x += [slice(start//2 + shift, -start//2 + shift) if start else slice(start, None) for start, shift  in zip(starts[:-1], start_shifts[:-1])]
```

Comment:
This is becoming really convoluted. I wonder if we can simplify it.

Reply:
How does the new revision look?